modules/table/src/main/java/org/apache/ignite/internal/table/AbstractTableView.java

```diff
@@ -68,15 +71,18 @@ protected <T> T sync(CompletableFuture<T> fut) {
     /**
      * Converts an internal exception to a public one.
      *
-     * @param th Internal exception.
+     * @param t Internal exception.
      * @return Public exception.
      */
-    protected IgniteException convertException(Throwable th) {
-        if (th instanceof IgniteException) {
-            return (IgniteException) th;
+    protected IgniteException convertException(Throwable t) {
+        if (t instanceof IgniteException) {
+            return (IgniteException) t;
+        } else if (t instanceof IgniteInternalCheckedException || t instanceof IgniteInternalException) {
+            // Map internal errors to IgniteException with code internal_err.
```

Comment:
why you have two `else` block with `instanceof` instead just return `new IgniteException(Common.INTERNAL_ERR, t)`?